src/pytorch_lightning/trainer/connectors/data_connector.py

```diff
@@ -439,10 +442,10 @@ def _request_dataloader(self, stage: RunningStage) -> Union[DataLoader, List[Dat
         return dataloader
 
     @staticmethod
-    def _resolve_overfit_batches(dataloaders: Collection[DataLoader], mode: RunningStage) -> Collection[DataLoader]:
+    def _resolve_overfit_batches(dataloaders: EVAL_DATALOADERS, mode: RunningStage) -> EVAL_DATALOADERS:
```

Comment:
Isn't this also used with training dataloaders?

Reply:
good catch! let me update